torchtune/utils/_checkpointing/_checkpointer.py

```diff
@@ -384,6 +384,14 @@ def load_checkpoint(self) -> Dict[str, Any]:
 
         if self._model_type == ModelType.PHI3_MINI:
             converted_state_dict[utils.MODEL_KEY] = phi3_hf_to_tune(merged_state_dict)
+        elif self._model_type == ModelType.MISTRAL_REWARD:
+            merged_state_dict["lm_head.weight"] = merged_state_dict.pop("score.weight")
```

Comment:
It does seem very tempting to add this one line change here. But this will start to add up and make this checkpointer really hard to read.

I do need to redesign this component a bit. But in the mean time sn alternative here (granted it's more lines of code with some copy paste) would be to define a custom ```convert_weights.py``` within the mistral folder and define a new ```mistral_reward_hf_to_tune``` and ```mistral_reward_tune_to_hf``` functions which do this mapping (and any future changes you might need for reward modeling). An example is what we did for [```phi3```](https://github.com/pytorch/torchtune/blob/main/torchtune/models/convert_weights.py). As I mentioned, its more copy-paste code, but I can definitely see this function updating as you add more heads etc. And I'm afraid the checkpointer will bloat up because of it.

@ebsmothers let me know what you think about this.

Reply:
I think this makes sense to me.